classes/AddressChecksumCore.php

```diff
@@ -49,7 +49,7 @@ public function generateChecksum($address)
         try {
             $fields = $address->getFields();
         } catch (Exception $e) {
-            throw new Exception($e->getMessage());
+            throw new Exception($this->trans('Your address is incomplete, please update it.', array(), 'Shop.Notifications.Error'));
```

Comment:
In controllers, trans is always accessible but here, we don't have to do that.
You can use $this->getTranslator()->trans(...) if you want but,if we change the "original" trans function, we have to change here to and it's not the good way to do this.
Maybe, the best solution is to do as usual.